eclipse-theia/theia
# Decorate Plugin Trees

#### What it does


Fixes #9642, closes #9643

This PR

 - enables Problem and SCM decoration in plugin `TreeView`s. The code for that is based roughly on #9643.
 - introduces an `AbstractFileTreeDecorator` class with common methods for decorating tree views and refactors the `ScmNavigatorDecorator` and `ProblemDecorator` to inherit from that class.
 - moves special-cased code for the open editors widget decoration from the `ProblemDecorator` into the `NavigatorOpenEditorsWidget` class
 - moves special handling of 'dot decorations' (indicating decorations propagated to a parent from a child) from the navigator into the `TreeWidget` class.

#### How to test


1. Follow the steps outlined in #9643
2. Modify a file in your workspace and save it.
3. Observe that git decoration colors are carried over to the caption in the navigator, and to the caption and the caption suffix in the Open Editors widget.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
